include/boost/graph/vf2_sub_graph_iso.hpp

```diff
@@ -75,6 +75,14 @@ template < typename Graph1, typename Graph2 > struct vf2_print_callback
 
 namespace detail
 {
+    template <typename Graph>
+    typename graph_traits<Graph>::vertices_size_type get_num_vertices(Graph& g)
+    {
+        typename graph_traits<Graph>::vertices_size_type N = 0;
+        BGL_FORALL_VERTICES_T(v, g, Graph)
+            N++;
+        return N;
```

Comment:
```suggestion
        typedef typename graph_traits<Graph>::vertex_iterator iter;
        std::pair<iter, iter> vs = vertices(g);
        return std::distance(vs.first, vs.second);
```
The old ``num_vertices()`` takes constant time, so having a raw loop which forces linear time is not ok. With ``std::distance()`` it should only take linear time when needed (e.g., for ``filtered_graph``).

Reply:
This is a good suggestion!